### DISCUSSION TITLE
ability beyond the page

### DISCUSSION THREAD

**Discussion by memory660** [stalniy/casl#816] [Q&A]
I'm a beginner and I don't know how to manage ability in a particular way
i use Angular with CASL

```
                                     role1                  role2

page1                             access               access
  show icone1                 read                  read
  show icone2                 write                  -
  show icone3                 read                  read
page2                               -                      read
  ...
```

```js
_pureAbility = inject(PureAbility);

let abilities = [];
if (role1) {
  abilities.push({ action: 'read', subject: page1});   // ?????????????? how to do
}
if (role2) {
  abilities.push({ action: 'read', subject: page1});  // ????????????   how to do
}
this._pureAbility.update(new PureAbility(abilities).rules);
```

```html
<div *ngIf="('create' | able: page1)">
```

how can i manage? please
because it's not just the page that's in "read" or "write" mode, but the icon elements that are.

**Comment by stalniy** [comment]
In this case your actual subject type is page name + icon name

but I guess this is just how you think about it. Very likely there is some Entity, who owns this icons (states?) and which information is displayed on that page, it’s also fine that there are multiple entities displayed on a single page and permissions should be defined for those entities

Subject type ideally should be taken from Domain layer not from Presentation

**Comment by memory660** [comment]
ok thanks, now I understand how to do it
